### PR TITLE
chore: reduce SauceLabs + Edge flakiness

### PR DESCRIPTION
**Pull request by jelbourn**
The changes in this PR _reduce_ the flakiness of Edge on SauceLabs but does not entirely eliminate it. 

* Remove unnecessary use of `async` and `fakeAsync`
* Inline resources for `gulp test:single-run` and `gulp e2e`
* Add `:watch:components:spec` task, and use that for `gulp test`, which fixes the watch behavior (but only if you're running only tests).

### CODE REVIEW

File: tools/gulp/tasks/unit-test.ts [hunk 1]
```diff
@@ -33,7 +37,10 @@ gulp.task('test', [':test:deps'], (done: () => void) => {
     configFile: path.join(PROJECT_ROOT, 'test/karma.conf.js')
   }, done).start();
 });
+
 gulp.task('test:single-run', [':test:deps'], (done: () => void) => {
+  inlineResources([DIST_COMPONENTS_ROOT]);
```
Comment (DevVersion) [inline tools/gulp/tasks/unit-test.ts]: Should this be replaced with the new gulp task you created?

**Comment by jelbourn** [comment]
R: @hansl

**Comment by hansl** [comment]
LGTM, but please fix the nit pointed by @DevVersion

File: tools/gulp/tasks/unit-test.ts (hunk 1 above)
Comment (jelbourn) [inline tools/gulp/tasks/unit-test.ts]: Done.

**Pull request merged**